src/python/intp_handler.py

```diff
@@ -0,0 +1,21 @@
+import code
+import io
+from code import InteractiveInterpreter
+from contextlib import redirect_stdout, redirect_stderr
+
+from pyspark import SparkConf, SparkContext
+
+class IntpHandler:
```

Comment:
I'm looking at the pyspark shell code (https://github.com/apache/spark/blob/master/python/pyspark/shell.py) -- does it work if you just call `exec(code_string)` directly, instead of using an `InteractiveInterpreter`? Since it's Python, calling `exec` should just run the user's code.

Reply:
Good point, that should work. I was unsure if `exec` uses the state from the main Python program but it looks like that is the case: https://www.programiz.com/python-programming/methods/built-in/exec